slidingwindow.h

```diff
@@ -1745,4 +1745,235 @@ void ConvolutionInputGenerator_1D_parallel(
 } // End generator
 
 
+/**
+* \brief Sliding Window unit that produces output vectors for feeding
+* a Matrix_Vector_Activate_Batch, implementing the im2col algorithm. To be used for single dimensional feature maps
+*
+* \tparam ConvKernelDim_x    	Dimension of the convolutional kernel - x axis
+* \tparam IFMChannels      	Number of Input Feature Maps
+* \tparam Input_precision  	Number bits per pixel
+* \tparam IFMDim_x          	Width of the Input Feature Map
+* \tparam OFMDim_x           	Width of the Output Feature Map
+* \tparam SIMD             	Number of input columns computed in parallel
+* \tparam Stride_x           	Stride of the convolutional kernel - x axis
+* \tparam R          	  		Datatype for the resource used for FPGA implementation of the SWG  - safely deducible from the parameters
+*
+* \param in                	Input stream
+* \param out               	Output stream
+* \param numReps           	Number of time the function has to be repeatedly executed (e.g. number of images)
+* \param r			  			Resource type for the hardware implementation of the memory block
+*/
+template<unsigned int ConvKernelDim_x,
+	unsigned int IFMChannels,
+	unsigned int Input_precision,
+	unsigned int IFMDim_x,
+	unsigned int OFMDim_x,
+	unsigned int SIMD,
+	unsigned int Stride_x,
+	typename R>
+void ConvolutionInputGenerator_1D(stream<ap_uint<SIMD*Input_precision> > & in, stream<ap_uint<SIMD*Input_precision> > & out, const unsigned int numReps, R const &r) {
+		CASSERT_DATAFLOW(IFMChannels % SIMD == 0);
+
+		const unsigned int multiplying_factor = IFMChannels/SIMD;
+		ap_uint<SIMD*Input_precision> inputBuf[IFMDim_x * multiplying_factor];
```

Comment:
this seems to be a simplified SWG that buffers the entire input before starting the generate the sliding window outputs. this will have some implications on:
- the buffer size for the SWG (proportional to the spatial size of the input, IFMDim_x)
- the latency (since all inputs must be read before any output generation starts)
- FIFO sizing (since input reading stops while generating outputs, must have large FIFO buffer before SWG in order to not stall the previous layer)

The non-square 2D SWG variants had buffer allocation like this:
```
const unsigned int number_blocks = ConvKernelDim_y/Stride_y + 1 ;
ap_uint<SIMD*Input_precision> inputBuf[number_blocks][Stride_x * IFMDim_x * multiplying_factor];
```
If using this non-square 2D SWG variant with a 1D input, what are the buffer size implications of putting the 1D spatial dimension on the X axis vs the Y axis? Do any of them perform better than the simplified 1D SWG here?